Skip token changes on non-hex maps

BoltsJ opened the pull request:
This skips token changes and goes back to the default settings for
gridless and square maps since it doesn't make sense to apply the
changes there.

BoltsJ committed:
Skip token changes on non-hex maps

This skips token changes and goes back to the default settings for
gridless and square maps since it doesn't make sense to apply the
changes there.

Ourobor commented:
This looks good. I'm currently navigating a massive mess I left myself when I last updated(I don't know why I thought setting the package manifest to a tagged url would be a good idea, but here we are :|), but I'll get this in asap.

BoltsJ commented:
I'm actually considering rebasing this on #18 . Were you planning on integrating that one soon?

Ourobor commented:
Probably. I still need to test it before I can accept it.
The **_dream_** for today is to get this, that pr. libwrapper implementation, and some new config options bundled together into a cohesive, proper update. But then I discovered this mess I need to untangle so people's copies will update properly so we'll see how much I get done.

Ourobor commented:
This is what I get for not having a dev branch >.>;

BoltsJ commented:
If you want, I have a github workflow that automatically creates a release archive and puts it in the github releases. It's adapted from an older league of foundry dev template.

Ourobor commented:
Maybe, but I don't typically need to update much. I'm also woefully out of touch with github and workflows <.<

Most of this mess is just planning out my moves very carefully before I execute because I have to commit the heresy that is modifying a tag and I'd like to not mess that up too.

BoltsJ commented:
I believe foundry has the ability to point to a new manifest when you publish the release there. The updater will see that the newest release on the foundry db has a different manifest url and prompt the user to change manifests iirc. It's used mostly for abandoned packages changing owners, but I believe you can get around this without editing tags.

Ourobor commented:
But black magic heresy evil...

Alright, that's fair. I wasn't sure how robust that was. I knew it asks for the package's manifest when you create a new version, but I wasn't sure if that was just used when you fresh installed. I think I'll work on the work I wanted to get done, try to just modify the manifest in the admin panel and see if the package will update properly and if not I can resort to evil stuff.

I'm not really sure if I should merge this pr now, or check on libruler and see if it's good too, so you to rebase on. I'm going to predict that there will be a conflict on the ruler stuff because of how dramatically libruler changes stuff, but other than that it'll probably be good. I guess it's up to you if you want to fix that with your rebase or let me figure it out

BoltsJ commented:
Im not busy today, so im totally fine with rebasing

BoltsJ commented:
Upon rebasing, I noticed that the ruler changes aren't working in the libruler branch. Do you want me to try to fix it, or do you just want to merge this as is?

Ourobor commented:
Your changes or theirs?

BoltsJ commented:
Theirs, as far as I can tell. I'm investigating and after checking out their branch, the ruler was still acting strange.

Update: the problem is with libruler itself. Disabling libruler fixes it.

Ourobor commented:
Yeah, there's an issue I'm seeing where it doesn't seem to adjust the snapping point for even sizes correctly. Gonna shift over to that pr with discussion of findings

BoltsJ committed:
Use built in canvas.grid.isHex

Instead of rolling a test for if a grid is a hex grid, just test
canvas.grid.isHex. This resolves an incompatibility with terrainRuler.

BoltsJ commented:
I'm going to close this as obsolete by https://github.com/Kuenaimaku/Hex-Size-Support/pull/1
